### PR TITLE
[Feature] - Update Edit Template page, including Publish functionality, and hook up to backend

### PR DESCRIPTION
**Pull request by jupiter007**
## Description

- Updated the Edit Template page to get data from the backend. You need the backend updates ([#153](https://github.com/CDLUC3/dmsp_backend_prototype/issues/153)) to test with the frontend changes.
- Added translations
- Added unit tests
- Updated Question styles to only show 3 lines of the question, with an ellipsis
- Added mutations for createTemplateVersion and archiveTemplate
- Added query for templateQuery

Fixes # ([206](https://github.com/CDLUC3/dmsp_frontend_prototype/issues/206))

## Type of change
- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
This was tested manually, as well as with unit tests.

To test,
- Get `feature/JS-add-lastPublishedDate-to-templates-table` branch of dmsp_backend_prototype
- Start this frontend branch
- Log in with `super@example.com` and navigate to `/en-US/template/15`
- Note that the Sections and questions display
- Note that when you click "Publish Template" button, you can save a versionedTemplate with the specified "visibility" and "changelog"
- Note that when you click "Save Draft", that a versionedTemplate is created with a versionType of "Draft"

## Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I updated the CHANGELOG.md and added documentation if necessary
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have completed manual or automated accessibility testing for my changes
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules (the dependent Backend changes need to be merged first)

### CODE REVIEW

File: hooks/useFormatDate.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { useFormatter } from 'next-intl';
```
Comment (jupiter007) [inline hooks/useFormatDate.ts:1]: Created this new hook to use the `useFormatter` from `next-intl` to format dates

File: i18n/request.ts [hunk 2]
```diff
@@ -14,11 +14,13 @@ export default getRequestConfig(async ({ requestLocale }) => {
   const mainMessages = (await import(`@/messages/${locale}/en-US.json`)).default;
   const templateMessages = (await import(`@/messages/${locale}/templates.json`)).default;
   const errorMessages = (await import(`@/messages/${locale}/errors.json`)).default;
+  const messaging = (await import(`@/messages/${locale}/messaging.json`)).default;
```
Comment (jupiter007) [inline i18n/request.ts:17]: Added a new `.json` file for messaging

**Review by briri — APPROVED** [review]
LGTM @jupiter007 I just had one comment about using `any`

File: app/[locale]/template/[templateId]/page.tsx [hunk 3]
```diff
@@ -18,272 +21,335 @@ import {
...
+        latestPublishVersion: data.template.latestPublishVersion ?? null,
+        latestPublishDate: formatDate(data.template.latestPublishDate) ?? null,
+        created: data.template.created ?? null,
+        /*eslint-disable @typescript-eslint/no-explicit-any*/
+        sections: data.template.sections?.map((section: any) => ({
+          id: section.id ?? null,
+          name: section.name ?? '',
+          displayOrder: section.displayOrder ?? null,
+          /*eslint-disable @typescript-eslint/no-explicit-any*/
+          questions: section.questions?.map((question: any) => ({
```
[the remark this answers is not in the document]
Comment (jupiter007) [reply app/[locale]/template/[templateId]/page.tsx]: Thanks @briri. I followed your suggestion and used the types from graphql.tsx, and had to filter out the null values to stop some errors from appearing.

**Pull request merged**